agent/engine/task_manager.go

```diff
@@ -472,19 +474,31 @@ func (mtask *managedTask) handleResourceStateChange(resChange resourceStateChang
 		return
 	}
 
+	defer mtask.engine.saver.Save()
+	// Set known status regardless of error so the applied status can be cleared. If there is error,
```

Comment:
The applied status can be cleared is bcuz updateAppliedStatusUnsafe is called in the SetKnownStatus? If this is the case, I think not all task resource's SetKnownStatus function calls updateAppliedStatusUnsafe, like dockerVolume. Did we already update that?

Reply:
good catch. i thought i checked every one of them. but apparently dockerVolume does not call it. double checking and that's the only missing one. will fix

Reply:
Fixed